app/api/packs/campaign.py

```diff
@@ -40,10 +40,11 @@ async def _section_profiles(self, request):
         objs = await self.data_svc.locate('objectives', match=access)
         platforms = dict()
         for a in abilities:
-            if a.platform in platforms:
-                platforms[a.platform].add(a.executor)
-            else:
-                platforms[a.platform] = set([a.executor])
+            for executor in a.executors:
+                if executor.platform in platforms:
+                    platforms[executor.platform].add(executor.name)
+                else:
+                    platforms[executor.platform] = set([executor.name])
```

Comment:
Seems like a `defaultdict(set)` could be helpful here, esp since you are building up another dictionary after which converts the sets to lists (also, do you know why we do that?)

Reply:
Guessing that the sets are converted to lists so `platforms` will be a dictionary with list values when converted to a JavaScript dictionary with Jinja's `to_json`. Unsure what will happen if a `set()` is passed through to JavaScript, but I will test.

As a note: This is building a list of platforms and associated executors that will be available in the GUI. This restricts users creating abilities through the GUI, though. We _should_ allow users to create arbitrary platforms and executors, rather than restrict the dropdown to the platforms and executors used in other abilities.

Reply:
Sounds good. It wasn't a blocking request and just thought it might simplify things (it might not though, heh)